### PR TITLE
Pass game data from lobby to input password

### PR DESCRIPTION
**Pull request by tamari-gray**
part of #60

### CODE REVIEW

File: test/create_game_section/Input_password/input_password_widget_test.dart [hunk 1]
```diff
@@ -24,29 +26,35 @@ void main() {
...
-      final mockGame = Game(
-          id: 'mock_game_123',
-          name: null,
-          creatorName: null,
-          sonarIntervals: null,
-          password: 'test_password',
-          boundarySize: 0,
-          location: Location(latitude: 0, longitude: 0),
-          phase: null);
+      final _mockGameService = MockGameService();
```
Comment (tamari-gray) [inline test/create_game_section/Input_password/input_password_widget_test.dart:29]: fixing broken tests, no new implementation

File: test/create_game_section/Lobby/lobby_widget_test.dart [hunk 2]
```diff
@@ -29,15 +33,26 @@ void main() {
...
 
       _controller.add(mockCreatedGames);
-      final mockDatabaseService = MockDatabaseService(controller: _controller);
 
       // init lobby page
       await tester.pumpWidget(
-        MultiProvider(providers: [
-          Provider<DatabaseService>.value(value: mockDatabaseService),
-        ], child: MaterialApp(home: LobbyScreen())),
+        MultiProvider(
```
Comment (tamari-gray) [inline test/create_game_section/Lobby/lobby_widget_test.dart:44]: same here, just fixing a test that was broken after i made changes to code

**Review by tamari-gray — COMMENTED** [review]
main purpose of this pr is for after a user joins a game, that game object can be accessed by proceeding routes, (input password and waiting screen)

**Review by nickmeinhold — APPROVED** [review]
Looks awesome mate!  I kinda just skimmed the tests again caus I trust your testing abilities... if you're cool with that I completely approve!

**Comment by tamari-gray** [comment]
Thank you! Yeah that's fine!

**Pull request merged**